strongbox-web-core/src/main/java/org/carlspring/strongbox/web/RepositoryMapping.java

```diff
@@ -0,0 +1,25 @@
+package org.carlspring.strongbox.web;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * This annotation refers to a controller method parameter related to a Repository which is going to be validated
+ * for provided path variables.
+ *
+ * @author Pablo Tiraoo
+ */
+@Documented
+@Retention(RUNTIME)
+@Target(PARAMETER)
+public @interface RepositoryMapping
+{
+
+    String storageVariableName() default "storageId";
```

Comment:
Thanks to this change you probably can use this approach in the `MavenArtifactController#copy` twice, for `srcRepository` and for `destRepository`